### PR TITLE
[METAED-1514] typos in metaed settings "API source directory" should be causing deploy to fail but are not

### PR DESCRIPTION
**Pull request by CSR2017**

### CODE REVIEW

File: src/client/LanguageClient.ts [hunk 1]
```diff
@@ -84,19 +85,50 @@ async function addSubscriptions(context: ExtensionContext) {
...
+          await showErrorNotification('To deploy, set Ods Api Deployment Directory under File -> Preferences -> Settings.');
+          return;
+        }
+        if (!isValidDirectory(deployDirectoryPath)) {
+          await showErrorNotification(
+            'Directory path not found, set proper Ods Api Deployment Directory under File -> Preferences -> Settings.',
+          );
+          return;
+        }
+        if (drivePattern.test(deployDirectoryPath) && !deployDirectoryPath.substring(0, 3).endsWith(path.sep)) {
```
Comment (bradbanister) [inline src/client/LanguageClient.ts]: I think you can remove this. It's covered by last check, and isn't cross-platform. The original ticket had an issue with "C:" versus "C:\\", but while strange is actually expected behavior. [Omitting the back slash means current directory on that drive](https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file#fully-qualified-vs-relative-paths), not the drive root directory (!)

Comment (CSR2017) [reply src/client/LanguageClient.ts]: @bradbanister 
If I remove that check the following use case is failing : Have "EdFi-ODS and Implementation" folders on C drive directly. On extension setting just give "C:" for Deploy directory, the deployment is running. But not creating any folders under "EdFi-ODS"(no artifacts getting created).

File: src/client/LanguageClient.ts [hunk 2]
```diff
@@ -88,15 +98,42 @@ async function addSubscriptions(context: ExtensionContext) {
...
+          await showErrorNotification('To deploy, set Ods Api Deployment Directory under File -> Preferences -> Settings.');
+          return;
+        }
+        if (!isValidDirectory(deployDirectoryPath)) {
+          await showErrorNotification(
+            'Directory path not found, set proper Ods Api Deployment Directory under File -> Preferences -> Settings.',
+          );
+          return;
+        }
+        if (process.platform.indexOf('win') !== -1) {
```
Comment (bradbanister) [inline src/client/LanguageClient.ts]: I think you want "win32" for your check, like `process.platform === 'win32'`. There is no other string that means Windows.

See https://nodejs.org/api/process.html#process_process_platform

**Review by bradbanister — CHANGES_REQUESTED** [review]
One small change

**Review by bradbanister — APPROVED** [review]

**Pull request merged**